objectivec/ort_value_internal.h

```diff
@@ -9,9 +9,18 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ORTValue ()
 
-- (nullable instancetype)initWithCAPIOrtValue:(OrtValue*)CAPIOrtValue
-                           externalTensorData:(nullable NSMutableData*)externalTensorData
-                                        error:(NSError**)error NS_DESIGNATED_INITIALIZER;
+/**
+ * Creates a value from an existing C++ API Ort::Value and takes ownership from it.
+ * Note: Ownership is guaranteed to be transferred on success but not otherwise.
```

Comment:
nit: IIUC by wrapping it in Ort::Value the release is automatically handled, so in the case of failure we release here. Can we explicitly say that happens as it's a slightly opaque to require the reader to translate `but not otherwise` combined with the arg being `Ort::Value&&` into that end result.

Reply:
yes, the release is handled by Ort::Value. it's not necessarily the case that it will be released if this method fails. the move into `_value` only happens at the end.
https://github.com/microsoft/onnxruntime/blob/d628532a6209098bdc185bd063ed43bc64ce007b/objectivec/ort_value.mm#L144
if it fails earlier, the argument won't be moved from even though it is an rvalue reference.

that said, with a call site like this:
https://github.com/microsoft/onnxruntime/blob/d628532a6209098bdc185bd063ed43bc64ce007b/objectivec/ort_session.mm#L133-L135
it will be released on failure as the Ort::Value is a temporary. but that's up to the caller.

will also note there's some precedence of conditionally not moving from rvalue reference arguments. for example, [std::unordered_map::try_emplace](https://en.cppreference.com/w/cpp/container/unordered_map/try_emplace).

Reply:
I did not know that the std::move at the calling site did nothing if the function being called didn't do another std::move.